src/Raven.Server/SchemaValidation/SchemaValidatorCache.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Client.Exceptions;
+using Raven.Server.Documents;
+using Raven.Server.SchemaValidation.ErrorMessage;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server.Json.Sync;
+using Sparrow.Server.Logging;
+
+namespace Raven.Server.SchemaValidation;
+
+public class SchemaValidatorCache : IDisposable
+{
+    private readonly RavenLogger _logger;
+    private readonly (IDisposable Return, JsonOperationContext Value) _context;
+    private Dictionary<string, SchemaValidator> _schemaValidatorsPerCollection;
+    private bool _disabled;
+
+    public SchemaValidatorCache(DocumentsContextPool contextPool, RavenLogger logger)
+    {
+        _context.Return = contextPool.AllocateOperationContext(out _context.Value);
+        _logger = logger;
+    }
+
+    public void Update(Client.Documents.Operations.SchemaValidation.SchemaValidationConfiguration configuration)
```

Comment:
this method will be doing a lot of work often e.g. when anything else will change in the database record, I think it would be best to impl comparison methods and compare prev and current instance. If they are fine, no-op